src/pylint_odoo/checkers/odoo_addons.py

```diff
@@ -517,6 +525,15 @@ class OdooAddons(OdooBaseChecker, BaseChecker):
                 "help": "List of valid odoo versions separated by a comma.",
             },
         ),
+        (
+            "deprecated-odoo-model-methods",
+            {
+                "type": "csv",
+                "metavar": "<comma separated values>",
+                "default": set(),
```

Comment:
We generally use the default value in the config parameter

It is impossible since it depends on the odoo version, right?

So, maybe we should use a way to set the dict in the configuration file

Something similar to:
`deprecated_odoo_model_methods={'16.0':['method61','method62'],'17.0': ['method71','method72']}` or something similar

Also, consider that the method is deprecated not only in the current version set but >= current version set

I mean, this 16.0 method will be deprecated for 17.0 too

So, they should be configured to raise the error for all odoo versions >=16.0

I know it is a good practice for configuration files but we can not have new sections using the same class

But I'm opened to use another way

Consider that pre-commit-vauxoo uses only a configuration file for all odoo versions.

Reply:
A dictionary is already being used:

```python
DFTL_DEPRECATED_ODOO_MODEL_METHODS = {"16.0": {"fields_view_get"}}
```

The value for deprecated methods is set on `open` if there was no config provided. I left the options as a CSV because I don't think most people will use it, I see it as a escape hatch, if you are going out of your way to use it, then you need to type out the whole list of methods to consider.

Reply:
The global variable is a dict

But using the configuration file it is not a dict

I mean, the following line only support a list of methods but they could be not valid for different versions of Odoo

```python
            self._deprecated_odoo_methods = set(self.linter.config.deprecated_odoo_model_methods)
```

For consistency, IMHO we should use dict for configuration file too

Or using set for global variable but not 2 ways to receive the values

If the user needs to define methods deprecated for 16.0 but not for 15.0 it should be able to be configured from configuration file or using the --parameters

Reply:
I think using set is an implementation detail. I use it because it seems to me like the best data structure for the task, a dictionary or list would be inferior for doing `x is in y` type of operations. 

Regarding configuration, I think the following could make us all happy:

```
[ODOOLINT]
deprecated_odoo_model_methods_versions_exclude = list of versions to exclude
```

This configuration lets you cover the following case:

> If the user needs to define methods deprecated for 16.0 but not for 15.0 it should be able to be configured from configuration file or using the --parameters

You could simply do:

`exclude_deprecated_odoo_model_methods_versions = 15.0`

This will exclude version 15.0 methods from the check, but still include methods from all other `valid_odoo_versions`
We could also add one more setting to exclude specific messages regardless of version, and that could be a list as well.

This way we have the most flexibility while keeping the defaults sane and useful for most users.